File: nestjs/src/orders/order.entity.ts
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
import { BaseTimeEntity } from '../common/base-time.entity';

@Entity({ name: 'orders_nest' })
export class OrderEntity extends BaseTimeEntity {
  @PrimaryGeneratedColumn({ type: 'bigint' })
  id!: number;

  @Column({ name: 'product_name', nullable: false })
  productName!: string;

  @Column({ type: 'integer', nullable: false })
  quantity!: number;

  @Column({ type: 'bigint', nullable: false })
  price!: number;

  constructor(partial?: Partial<OrderEntity>) {
    super();
    if (partial) {
      Object.assign(this, partial);
    }
  }
  // nestjs 관례에 맞게 springboot랑 차이 둠
  // entity 인스턴스는 생성자보다 orm에 의존함 (repository.create())
}

File: springboot/springboot/src/main/java/com/github/seungwoo/springboot/order/OrderEntity.java
package com.github.seungwoo.springboot.order;

import com.github.seungwoo.springboot.common.BaseTimeEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "orders_spring")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class OrderEntity extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "product_name", nullable = false)
    private String productName;

    @Column(name = "quantity", nullable = false)
    private Integer quantity;

    @Column(name = "price", nullable = false)
    private Long price;

    @Column(name = "hash_id", nullable = false, length = 128)
    private String hashId;

    private OrderEntity(String productName, Integer quantity, Long price, String hashId) {
        this.productName = productName;
        this.quantity = quantity;
        this.price = price;
        this.hashId = hashId;
    }

    public static OrderEntity of(String productName, Integer quantity, Long price, String hashId) {
        return new OrderEntity(productName, quantity, price, hashId);
    }
}
